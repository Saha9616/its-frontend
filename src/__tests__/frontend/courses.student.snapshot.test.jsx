import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import MyCourseView from "@/app/courses/page";

// mock server action getUserProps
jest.mock("@/actions/getUserProps", () => ({
  getUserProps: jest.fn().mockResolvedValue({
    id: "mockId",
    email: "mockEmail",
    password: "mockPw",
    role: "STUDENT",
    school_id: "mockSchoolId",
    created_courses: [],
    joined_courses: [
      {
        id: "mockCourseId",
        code: "mockCourseCode",
        name: "mockCourseName",
        creator_id: "mockCreatorId",
        school_id: "mockSchoolId",
      },
    ],
  }),
}));

it("renders student's courses page unchanged", async () => {
  const result = await MyCourseView();
  const { container } = render(result);
  expect(container).toMatchSnapshot();
});
