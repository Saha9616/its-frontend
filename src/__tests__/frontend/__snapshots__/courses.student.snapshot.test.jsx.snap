// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders student's courses page unchanged 1`] = `
<div>
  <div
    class="flex justify-between items-center"
  >
    <h1
      class="text-xl font-semibold"
    >
      My Courses
    </h1>
  </div>
  <div
    class="mt-4 p-4 border rounded-md min-h-[calc(100vh-160px)]"
  >
    <div
      class="grid grid-cols-4 md:grid-cols-6 gap-4"
    >
      <a
        href="/courses/mockCourseCode"
      >
        <div
          class="rounded-lg border bg-card text-card-foreground shadow-sm h-40"
        >
          <div
            class="flex flex-col space-y-1.5 p-6"
          >
            <h3
              class="text-2xl font-semibold leading-none tracking-tight flex items-center space-x-2"
            >
              <span>
                mockCourseCode
              </span>
            </h3>
          </div>
          <div
            class="p-6 pt-0 space-y-2"
          >
            <h4
              class="font-medium"
            >
              mockCourseName
            </h4>
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
